components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1494,4 +1411,454 @@ private static Validator resourceToValidatorObject(
+            if (CertificateMgtErrors.ERROR_CERTIFICATE_DOES_NOT_EXIST.getCode().equals(e.getErrorCode())) {
+                throw new CertificateValidationException
+                        ("Certificate with the id: " + certificateId + " does not exist.", e);
+            } else {
+                throw new CertificateValidationException("Error when getting certificate from certificate manager.", e);
+            }
+        }
+    }
+
+    private static CertObject processCertificateInStore(String tenantDomain, X509Certificate certificate,
+                                                        String certificateId, boolean isUpdate)
+            throws CertificateValidationException {
+
+        try {
+            org.wso2.carbon.identity.configuration.mgt.core.model.Resource resource = CertValidationDataHolder
+                    .getInstance().getConfigurationManager()
+                    .getResourceByTenantId(IdentityTenantUtil.getTenantId(tenantDomain),
+                            X509_CA_CERT_RESOURCE_TYPE, CERTS);
+            InputStream inputStream = getResourceFileInputStream(resource);
+            if (inputStream == null) {
+                return null;
+            }
+
+            IssuerDNMap issuerDNMap = ModelSerializer.deserializeIssuerDNMap(convertInputStreamToString(inputStream));
+            String issuerDN = getNormalizedName(certificate.getIssuerDN().getName());
+            String serialNumber = getNormalizedName(certificate.getSerialNumber().toString());
+
+            List<String> ocspUrls = getValidationUrls(certificate, tenantDomain, OCSP_VALIDATOR);
+            List<String> crlUrls = getValidationUrls(certificate, tenantDomain, CRL_VALIDATOR);
+
+            CertObject certObject = new CertObject();
+            certObject.setCertId(isUpdate ? certificateId : UUID.randomUUID().toString());
+            certObject.setSerialNumber(serialNumber);
+            certObject.setCrlUrls(crlUrls);
+            certObject.setOcspUrls(ocspUrls);
+
+            updateIssuerDNMap(issuerDNMap, issuerDN, serialNumber, certObject);
+
+            return saveUpdatedResource(resource, issuerDNMap);
+        } catch (ConfigurationManagementException | IOException e) {
+            throw new CertificateValidationException("Error while processing the resource", e);
+        }
+    }
+
+    private static List<String> getValidationUrls(X509Certificate certificate, String tenantDomain,
+                                                  String validatorType)
+            throws ConfigurationManagementException, CertificateValidationException {
+
+        List<String> urls = new ArrayList<>();
+        if (!isSelfSignedCert(certificate)) {
+            for (Validator validator : getValidatorsFromConfigStore(tenantDomain)) {
+                if (validator.isEnabled() && validatorType.equals(validator.getDisplayName())) {
+                    urls = validatorType.equals(OCSP_VALIDATOR) ? getAIALocations(certificate) :
```

Comment:
Use `OCSP_VALIDATOR.equals(validatorType)` to make it null safe

Reply:
fixed with https://github.com/wso2-extensions/identity-x509-commons/pull/57/commits/52f92488d265780c25c0d23c5ec638ca00bd3f59